### DISCUSSION TITLE
Compile IM with FFTW on Windows

### DISCUSSION THREAD

**Discussion by Waveluke** [ImageMagick/ImageMagick#2985] [Help]
So I have been trying to get FFT imagemagick in Windows, and I found this instruction sheet. http://www.imagemagick.org/discourse-server/viewtopic.php?f=4&t=14251#p56836

It seemed out of date, being a decade old, but I tried to follow it anyway. I had success until this step:

    Run the configure program. In the dialog, click Next.
    In the next dialog window, select "Edit magick-config.in".
    This will start up notepad and allows you to edit the file.
    Check that MAGICKCORE_QUANTUM_DEPTH is the required value (16 is best when using FFTW and/or HDRI) then check that 
    MAGICKCORE_HDRI_SUPPORT is defined.
    Finally, after HDRI, add this:
    #define MAGICKCORE_FFTW_DELEGATE

I could not find an option in the configure dialogue gui to edit “magick-config.in” despite being able to set the HDRI 16 bit mode enablement. Manual search for that the file “magick-config.in” proved fruitless, despite searching the entire repository and all files generated. Nonetheless, since I had done the linking to MS Visual Studio to the delegate library step earlier, I tried the compile anyway. Imagemagick was successfully compiled, and I continued to make trig function art with imagemagick but no fft filter support.

I tried to sign up to the imagemagick forum to ask this question, http://www.imagemagick.org/discourse-server/index.php?sid=3ff41626a0311a92e33eff2d622f3091 , but found no way to create an account required to post.

Anyone have an updated easy for people like me who only know intro level programming stuff tutorial, or pre-compiled windows 10 imagemagick fftw included binaries that they could private message share on Google drive with me?

Thanks,
Waveluke.

**Comment by Ivan1923stop** [comment]
For LInux they do requests quickly.

1. [1](https://dev.getsol.us/T9403)
2. [2](https://imagemagick.org/script/changelog.php)
(enable FFTW support with --with-fftw configure script command-line option.)

**Comment by snowbie** [comment]
Have you figured this out? I was able to successfully build IM after a few attempts, but I was never able to incorporate fftw/fftw3 in any of the builds. I was able to find an old binary from 2012 that included fftw, but I'd prefer to have it included in a fresh build with all of my other delegates enabled.

**Comment by snibgo** [comment]
The Windows source code at https://imagemagick.org/script/install-source.php#windows includes fftw. I build IM using Cygwin tools, but other methods are available.

**Comment by Ivan1923stop** [comment]
What should I reply on tis?Sure I would like to have fftw and other people I think.Andrei,  02.10.2021, 22:40, "snowbie" ***@***.***>: Have you figured this out? I was able to successfully build IM after a few attempts, but I was never able to incorporate fftw/fftw3 in any of the builds. I was able to find an old binary from 2012 that included fftw, but I'd prefer to have it included in a fresh build with all of my other delegates enabled.—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.
